appcues/src/main/java/com/appcues/data/model/ExperienceStepFormState.kt

```diff
@@ -20,58 +20,52 @@ internal class ExperienceStepFormState {
     // this can be used by buttons needing to know if they are enabled or not based on required input
     val isFormComplete = mutableStateOf(true)
 
-    fun getValue(primitive: TextInputPrimitive) =
-        getItem(primitive).text
+    fun register(primitive: ExperiencePrimitive) {
```

Comment:
registration of primitives with the formState object is now explicit.  Previously, it was implicitly done on the first call to get or set the value - a side effect. Making it explicit makes it simpler to test, for one, and feels generally a better approach to me.  It only adds a line atop each of the two primitives that participate in form state (textInput and optionSelect) - and simplifies this class quite a bit.